docs/guide.ipynb

```diff
@@ -0,0 +1,191 @@
+     ]
+    }
+   ],
+   "source": [
+    "from draco import answer_set_to_dict, run_clingo\n",
+    "from draco.programs import define, hard, helpers, constraints, generate\n",
+    "from pprint import pprint\n",
+    "\n",
+    "prog = (\n",
+    "    generate.program\n",
+    "    + define.program\n",
+    "    + helpers.program\n",
+    "    + hard.program\n",
+    "    + constraints.program\n",
+    ")\n",
+    "\n",
+    "scatter = \"\"\"\n",
+    "    attribute(number_rows,root,100).\n",
+    "\n",
+    "    entity(field,root,(f,0)).\n",
+    "    attribute((field,name),(f,0),temperature).\n",
+    "    attribute((field,type),(f,0),number).\n",
+    "\n",
+    "    entity(field,root,(f,1)).\n",
+    "    attribute((field,name),(f,1),precipitation).\n",
+    "    attribute((field,type),(f,1),number).\n",
+    "\n",
+    "    entity(view,root,(v,0)).\n",
+    "    entity(mark,(v,0),(m,0)).\n",
+    "    entity(encoding,(m,0),(e,0)).\n",
+    "    attribute((encoding,field),(e,0),(f,0)).\n",
+    "    entity(encoding,(m,0),(e,1)).\n",
+    "    attribute((encoding,field),(e,1),(f,1)).\n",
+    "\n",
+    "    entity(scale,(v,0),(s,0)).\n",
+    "    entity(scale,(v,0),(s,1)).\n",
+    "\n",
+    "    #show entity/3.\n",
+    "    #show attribute/3.\n",
+    "\"\"\"\n",
+    "\n",
+    "for model in run_clingo(prog + scatter, 1):\n",
+    "    pprint(answer_set_to_dict(model.answer_set))\n",
+    "    print(model.answer_set)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8866d38f",
+   "metadata": {},
+   "source": [
+    "If you see that there are too many recommendations, you can:\n",
+    " * add more hard constraints\n",
+    " * modify your generator and hard constraints to reduce symmetry in the search space (ex. similar recommendations with switched entity ids)\n",
+    "\n",
+    "\n",
+    "If you see too few recommendations, you can:\n",
+    " * check if some of your constraints are too tight, and move them to soft constraints\n",
+    " \n",
+    "\n",
+    "If you see no recommendations, you might have made mistakes in the hard constraints. You can allow violations to check what are the common ones by doing the following:"
```

Comment:
Maybe say what you can do and then point to the example. For example "You can allow violations to check what are the common ones by .... Below is an example:"